conans/client/cmd/search.py

```diff
@@ -57,13 +62,19 @@ def search_packages(self, ref=None, remote_name=None, query=None, outdated=False
     def _search_packages_in_local(self, ref=None, query=None, outdated=False):
         packages_props = search_packages(self._cache, ref, query)
         ordered_packages = OrderedDict(sorted(packages_props.items()))
+
         try:
             recipe_hash = self._cache.package_layout(ref).load_manifest().summary_hash
         except IOError:  # It could not exist in local
             recipe_hash = None
 
-        if outdated and recipe_hash:
+        if outdated:
             ordered_packages = filter_outdated(ordered_packages, recipe_hash)
+        elif self._cache.revisions_enabled:
+            # With revisions, by default filter the packages not belonging to the recipe
+            # unless outdated is specified.
+            metadata = self._cache.package_layout(ref).load_metadata()
+            ordered_packages = filter_by_revision(metadata, ordered_packages)
```

Comment:
I can have `outdated=True` (it is passed from the command line) and _revisions enabled_... 

I think that the `filter_by_revision` should be done always if _revisions enabled_ (we can have packages in the cache corresponding to older revisions of the recipe) and then we should filter the outdated ones if requested by the user (although it might not make sense with the revisions enabled).

Said that it is exactly what it is implemented 🤦‍♂️ ... unless the user forces the revision, or several revisions were created using the same commit (uncommitted changes).

Reply:
Not really sure about this after looking at the `filter_by_revision` function

Reply:
I think this is correct.

Reply:
me too, after reviewing the tests I think it works as expected.